### ISSUE TITLE
Players can be seen wearing different headgear after having been in a vehicle

### ISSUE THREAD

**Issue by handsm** [DarklightGames/DarkestHour#870]
This bug was pointed out to me by @dynamittt :

This can affect player roles that have a random headgear selection after a player has been in a vehicle and then exited. It is a multi-player bug that can happen if that player drops out of network relevancy to the observing player and later becomes net relevant again (causing the 'headgear player' to re-spawn on the observing player's computer).

The problem is that a new random headgear selection is being made by the server every time a player exits a vehicle and re-possesses their player pawn. A changed headgear choice gets replicated to all net clients, but has no immediate effect. However, if another player drops out of and then back into net relevancy, their player pawn re-spawns and you see any changed headgear selection.

Reproduction steps (won't work every time because a typical 50/50 random choice often returns the same headgear choice as before):

1. Player A chooses a player role with a random headgear choice (e.g. US rifleman on target range map - may or may not have netting over helmet).
2. Player B observes Player A and notes his headgear.
3. Player A gets into and out of a vehicle (no visible change yet).
4. Player B moves so that player A becomes no longer net relevant to player B.
5. Player B moves so that player A becomes relevant again.
6. Player B will see a new random choice of headgear (may or may not be the sane as before).

The sequence of steps 3 and 4 isn't important. Also it doesn't matter which player moves to create the network relevancy drop out.

Note - here are two quick ways to guarantee a drop out of network relevancy:
- Disconnect from server, reconnect and re-spawn (player B would do that in this example).
- On target range map, switch to other team, wait five seconds, then switch back (again, Player B). The huge dividing wall on this map is BSP that causes a network drop.

**Comment by handsm** [comment]
I have fixed this and will commit soon.

Note that in single player this flawed functionality actually spawns a new headgear actor every time a player gets out of a vehicle. Only the most recent is recorded as the pawn's Headgear actor reference and gets destroyed when the pawn is destroyed. Any earlier headgear actors just accumulate in the level!

**Comment by handsm** [comment]
Fixed in commit https://github.com/DarklightGames/DarkestHour/commit/fee4cf592fdeb99f6cd314a531485db478e3e480.

Headgear selection, replication and spawning is quite flawed in several ways, but this is a simple one. We already have a bHasBeenPossessed bool in DHPawn, which gets flagged when the pawn is first possessed, to stop it repeating things it shouldn't. So I've simply moved the headgear selection functionality under that check.

Also did the same for getting/setting the player's ammo pouch and detached limb classes, as this also only needs to be done once, although in that case it's just an optimisation and not a bug fix, as there's no randomisation of ammo pouches or limbs.

Also moved spawning headgear and ammo pouch actors under bHasBeenPossessed to stop standalones and listen servers from re-spawning duplicate headgear and ammo pouch actors every time player exits a vehicle (& so repossesses his player pawn).

Re-ordered the function a little for logical flow and connection, and added better locals for simplicity and clarity. Deleted setting bUsedCarriedMGAmmo to false as it's pointless, as it will be false anyway when the pawn spawns.
